### DISCUSSION TITLE
Issue with oracle connection while using thik mode

### DISCUSSION THREAD

**Discussion by ajit-satav** [oracle/python-oracledb#363] [General]
I am using oracledb package along with instantclient_23_3 to connect oracle database with thik mode.  Below it he sample code for same.  oracle_config is the local directory which contains tnsnames.ora and sqlnet.ora

sqlnet.ora - 
WALLET_LOCATION=(SOURCE=(METHOD=FILE)(METHOD_DATA=(DIRECTORY=/<LOCAL_DIR_PATH>/oracle/wallet)))

tnsnames.ora -
oracle_qa =
(DESCRIPTION =
   (ADDRESS = (PROTOCOL = TCPS)(HOST = <HOST>)(PORT = <PORT>))
   (CONNECT_DATA =
     (SERVER = DEDICATED)
     (SERVICE_NAME = <service_name>)
   )
 )

Python file - 
oracledb.init_oracle_client(lib_dir="/<PATH>/instantclient_23_3",config_dir = "oracle_config")
connection = oracledb.connect(
            user="<USER>",
            password="<PASSWORD>",
            dsn="oracle_qa"
        )
 if not connection:
        return {"error": "Failed to connect to Oracle database"}
else:
       return "connection successful"

With this code while establishing connection, on line oracledb.connect program terminates without any error, nothing gets executed after this line and connection is not getting established. Do we see any issue with above code? what could be possible reasons for program getting terminated without any error.

**Comment by ajit-satav** [comment]
resolved. it was issue with local directory access

**Reply by cjbj** [reply]
@ajit-satav  Could you please share more information because I would like to reproduce and report this problem to the Instant Client team.  What access were you trying to make?
